pyiron_atomistics/sphinx/output_parser.py

```diff
@@ -417,3 +417,89 @@ def results(self):
             if len(value) > 0:
                 results["dft"][key] = value
         return results
+
+
+class SphinxWavesParser:
+    """Class to read SPHInX waves.sxb files (HDF5 format)
+
+    Initialize with waves.sxb filename, or use load ()
+
+    """
+
+    def __init__(self, file_name="waves.sxb", cwd="."):
+        """
+        Args:
+            file_name (str): file name
+            cwd (str): directory path
+        """
+        self._eps = None
+        if Path(file_name).is_absolute():
+            self.wfile = h5py.File(Path(file_name))
+        else:
+            path = Path(cwd) / Path(file_name)
+            self.wfile = h5py.File(path)
+
+    @property
+    def _n_gk(self):
+        return self.wfile['nGk'][:]
+    
+    @property
+    def mesh(self):
+        return self.wfile["meshDim"][:]
+
+    @property
+    def Nx(self):
+        return self.wfile['meshDim'][0]
+
+    @property
+    def Ny(self):
+        return self.wfile['meshDim'][1]
+
+    @property
+    def Nz(self):
+        return self.wfile['meshDim'][2]
+
+    @property
+    def n_states(self):
+        return self.wfile["nPerK"][0]
+
+    @property
+    def n_spin(self):
+        return self.wfile["nSpin"].shape[0]
+
+    @property
+    def k_weights(self):
+        return self.wfile["kWeights"][:]
+
+    @property
+    def k_vec(self):
+        return self.wfile["kVec"][:]
+
+    @property
+    def eps(self):
+        """All eigenvalues (in Hartree) as (nk,n_states) block"""
+        if (self._eps is None):
+            self._eps = self.wfile['eps'][:].reshape(-1,self.n_spin,self.n_states)
+        return self._eps.T #change
+
+    def get_psi_rec(self, i, ispin, ik):
+        """Loads a single wavefunction on full FFT mesh"""
+        if i < 0 or i >= self.n_states:
+            raise IndexError(f"i={i} fails 0 <= i < n_states={self.n_states}")
+        if ispin < 0 or ispin >= self.n_spin:
+            raise IndexError(f"ispin={ispin} fails 0 <= ispin < n_spin={self.n_spin}")
+        if ik < 0 or ik >= self.nk:
+            raise IndexError(f"ik={ik} fails 0 <= ik < nk={self.nk}")
+
+        res = np.zeros(shape=self.mesh, dtype=np.complex128)
+        off = self._n_gk[ik] * (i + ispin * self.n_states)
+        psire = self.wfile[f"psi-{ik+1}.re"][off : off + self._n_gk[ik]]
+        psiim = self.wfile[f"psi-{ik+1}.im"][off : off + self._n_gk[ik]]
+        res.flat[self.wfile["fftIdx"][off : off + self._n_gk[ik]]] = psire + 1j * psiim
```

Comment:
this won't work, as fftIdx for a specific k inside the global fft_idx var in the hd5 file has different offsets than psi for a specific ispin, istate inside the k-specific psi-variable inside hd5 file.
I also would guess that keeping fft_idx in memory instead of reloading for each new psi is better. If not keeping all fftIdx at all time, at least caching would be desirable (there is a caching function decorator).